firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java

```diff
@@ -0,0 +1,565 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.model.serialization;
+
+import android.util.Base64;
+import android.util.JsonReader;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.firebase.crashlytics.internal.model.AutoCrashlyticsReportEncoder;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.CustomAttribute;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event;
+import com.google.firebase.crashlytics.internal.model.ImmutableList;
+import com.google.firebase.encoders.DataEncoder;
+import com.google.firebase.encoders.json.JsonDataEncoderBuilder;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CrashlyticsReportJsonTransform {
+
+  private static final DataEncoder CRASHLYTICS_REPORT_JSON_ENCODER =
+      new JsonDataEncoderBuilder()
+          .configureWith(AutoCrashlyticsReportEncoder.CONFIG)
+          .ignoreNullValues(true)
+          .build();
+
+  @NonNull
+  public String reportToJson(@NonNull CrashlyticsReport report) {
+    return CRASHLYTICS_REPORT_JSON_ENCODER.encode(report);
+  }
+
+  @NonNull
+  public String eventToJson(@NonNull CrashlyticsReport.Session.Event event) {
+    return CRASHLYTICS_REPORT_JSON_ENCODER.encode(event);
+  }
+
+  @Nullable
+  public CrashlyticsReport reportFromJson(@NonNull String json) {
+    try (JsonReader jsonReader = new JsonReader(new StringReader(json))) {
+      return parseReport(jsonReader);
+    } catch (IOException e) {
+      // TODO: Handle the IOException
+      return null;
```

Comment:
I don't love this, but I won't push back too hard on it. It's just harder to make sure you deal with null upstream than to deal with an Exception.

Reply:
Agreed, hence the TODO.